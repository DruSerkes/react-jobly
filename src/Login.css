.Login {
	padding-top: 10%;
}

.Login-Container {
	width: 40vw;
	margin: 0 auto;
}

.Login-Buttons {
	text-align: right;
	width: 80%;
	margin: 0 auto;
}

.Login-Buttons button {
	margin: 0;
}

.Login-Buttons .active {
	background-color: rgb(11, 11, 161);
}

.error {
	color: red;
	margin-bottom: 1.2rem;
}

.LoginForm,
.RegistrationForm {
	background-color: white;
	width: 80%;
	margin: 0 auto;
	border-radius: 10px;
	border: 1px solid lightgray;
	padding: 1rem;
	display: flex;
	justify-content: center;
	align-items: center;
}
