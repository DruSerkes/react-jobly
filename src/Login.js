import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import LoginForm from './forms/LoginForm';
import RegistrationForm from './forms/RegistrationForm';

// write a function to pass down to the forms for their handleSubmits to login/register a user

const Login = () => {
	const [ formView, setFormView ] = useState('login');
	const history = useHistory();

	const doLogin = () => {
		history.push('/');
	};

	return (
		<div className="Login">
			<div className="Login-Container">
				<div className="Login-Buttons">
					<button onClick={() => setFormView('login')}>Login</button>
					<button onClick={() => setFormView('register')}>Register</button>
				</div>
				{formView === 'login' ? <LoginForm doLogin={doLogin} /> : <RegistrationForm doLogin={doLogin} />}
			</div>
		</div>
	);
};

export default Login;
