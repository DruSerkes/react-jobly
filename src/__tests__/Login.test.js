import React from 'react';
import userContext from '../Context';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Login from '../Login';

describe('Login tests', () => {
	it('renders without crashing', () => {
		let currentUser = null;

		render(
			<userContext.Provider value={{ currentUser }}>
				<MemoryRouter>
					<Login />
				</MemoryRouter>
			</userContext.Provider>
		);
	});

	it('matches snapshot', () => {
		let currentUser = null;

		const { asFragment } = render(
			<userContext.Provider value={{ currentUser }}>
				<MemoryRouter>
					<Login />
				</MemoryRouter>
			</userContext.Provider>
		);
		expect(asFragment()).toMatchSnapshot();
	});

	it('shows a login form', () => {
		let currentUser = null;
		const { getByText } = render(
			<userContext.Provider value={{ currentUser }}>
				<MemoryRouter>
					<Login />
				</MemoryRouter>
			</userContext.Provider>
		);
		expect(getByText('Submit')).toBeInTheDocument();
	});
});
