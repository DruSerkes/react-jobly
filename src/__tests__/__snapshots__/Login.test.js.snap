// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login tests matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Login"
  >
    <div
      class="Login-Container"
    >
      <div
        class="Login-Buttons"
      >
        <button
          class="login active"
        >
          Login
        </button>
        <button
          class="register"
        >
          Register
        </button>
      </div>
      <div
        class="LoginForm"
      >
        <form
          action="#"
        >
          <label
            for="username"
          >
            Username
          </label>
          <input
            class="text-input"
            name="username"
            type="text"
            value=""
          />
          <label
            for="password"
          >
            Password
          </label>
          <input
            class="text-input"
            name="password"
            type="password"
            value=""
          />
          <button>
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
